addons/project/static/src/views/burndown_chart/burndown_chart_model.js

```diff
@@ -1,14 +1,78 @@
 /** @odoo-module **/
 
 import { GraphModel } from "@web/views/graph/graph_model";
+import { sortBy } from "@web/core/utils/arrays";
+
+export const SEP = " / ";
 
 export class BurndownChartModel extends GraphModel {
+    /**
+     * @override
+     */
+    setup(params) {
+        this.stageSeq = {};
+        super.setup(params);
+    }
+
+    /**
+     * Fetch the sequence of each stage in the project. This function alters this.stageSeq
+     * @protected
+     * @param {Object} searchParams
+     */
+    async _fetchStageSeq(searchParams) {
+        const data = await this.orm.webSearchRead(
+            "project.task.type",
+            [["project_ids", "in", searchParams.context.active_id]],
+            ["name", "sequence", "id"]
+        );
+
+        const dataClean = {};
+        data.records.forEach((obj) => {
+            dataClean[obj.id] = { sequence: obj.sequence, name: obj.name };
+        });
+        return dataClean;
+    }
+
+    /**
+     * @param {SearchParams} searchParams
+     */
+    async load(searchParams) {
+        this.searchParams = searchParams;
+        // if the stage_seq wasn't given by the action (for exemple if the page is simply reloaded)
+        if (!this.searchParams.context.stage_seq) {
+            this.stageSeq = await this._fetchStageSeq(searchParams);
+        } else {
+            this.stageSeq = this.searchParams.context.stage_seq;
+        }
+        return super.load(searchParams);
+    }
+
+    /**
+     * @override
+     */
+    async _prepareData() {
+        const processedDataPoints = this._getProcessedDataPoints();
+        this.data = this._getData(processedDataPoints);
+        this.data.datasets.forEach(
+            (ds) => {
+                // In case there was no identifiers in the datasets 
+                if (!ds.identifiers || ds.identifiers.size === 0) {
+                    ds.orderIndex = -1;
+                }
+                else {
+                    // In case the identifier doesn't correspond to any stage.id
+                    ds.orderIndex = this.stageSeq[Math.min(...ds.identifiers)] !== undefined ? this.stageSeq[Math.min(...ds.identifiers)].sequence : -1;
+                }
+            });
```

Comment:
```suggestion
        for (const dataset in datasets) {
            if (!dataset.identifiers || !dataset.identifiers.size) {
                // In case there was no identifiers in the datasets 
                dataset.orderIndex = -1;
            } else {
                // In case the identifier doesn't correspond to any stage.id
                const stage = this.stageSeq[Math.min(...dataset.identifiers)];
                dataset.orderIndex = stage !== undefined ? stage.sequence : -1;
            }
        });
```
- This way to iterate is preferred. (Or map if possible ?)
- `blabla.size === 0` is the same as `!blabla.size`.
- `} else {`
- `this.stageSeq[Math.min(...dataset.identifiers)]` is :vomiting_face:, let's write it only once.
- Comments on the same level.

Reply:
Also, min is pretty arbitrary... But I guess, it has to be since we have a Set.
@Polymorphe57 Thank you for your feedback! I'm not sure I understand your point. Could explain a little further (maybe wwith a usecase) please? :pray: